### ISSUE TITLE
Mac OS -need to come up with an option to switch to a specific node without using the disconnect button.

### ISSUE THREAD

**Issue by Denis-Asipenko** [BeamMW/beam-ui#213]
**Bug description**
If the wallet has stuck swap transactions in the status "in progress" and the user clicked the "disconnect" button in the settings of the electrum node, then when the wallet is closed, this leads to a wallet crash. We need to find a solution to switch to a specific node without disconnecting from the node. This should be a reconnection, not a disconnection. The wallet crash is only on Mac OS, but the solution will be for all systems.
We need:
1) Decide how it will be designed in the user interface.
2) When the design is ready to do the technical part.

**Comment by beam-mw** [comment]
In case swap is in progress, in the settings screen 'Disconnect' button should be removed, and replaced with 'Change server' button. 

CLicking this button will open a dialog with two options (just like in the settings) 'Random node' and 'Specific node'. 

In case of specific node fields for ip and port should be shown. 

The dialog should have a reconnect button at the bottom, After the server details are set, clicking on  the reconnect button performs disconnect from the previous server and reconnect to new one

**Comment by sasha-abramovich** [comment]
still in discussion with @beam-mw

**Comment by dariatarakanova** [comment]
**If 1 node lost connection:**

1) Wallet screen - Change wallet status to "connection to DASH/BITCOIN/etc node lost: 1.2.3.4" and change color to red also add button "reconnect" that leads to settings/node block https://zpl.io/VQk9pA5

2) Atomic Swap screen - https://zpl.io/V1Dl1l4

3) Settings screen - https://zpl.io/bJJrJeE (in the bottom)

**If 2+ nodes lost connection:**

1) Wallet screen - Change wallet status to "lost connection to nodes" and change color to red also add button "reconnect" that leads to settings/node block (first - to the first node in the list) https://zpl.io/aXerrkE

2) Atomic swap screen - https://zpl.io/agjDDwN

3)Settings screen - https://zpl.io/bJJrJeE (in the bottom).

Bottom line:
green status - all good
red status - 1 or more nodes not connected (lost connection)
yellow status - reconnecting
